scripts/nlvr/generate_data_from_erm_model.py

```diff
@@ -0,0 +1,73 @@
+#! /usr/bin/env python
+
+# pylint: disable=invalid-name,wrong-import-position
+import sys
+import os
+import json
+import argparse
+
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(os.path.join(__file__, os.pardir)))))
+
+from allennlp.data.dataset_readers import NlvrDatasetReader
+from allennlp.data.fields import MetadataField
+from allennlp.models.archival import load_archive
+from allennlp.semparse.worlds import NlvrWorld
+
+
+def make_data(input_file: str,
+              output_file: str,
+              archived_model_file: str,
+              max_num_decoded_sequences: int) -> None:
+    reader = NlvrDatasetReader(output_agendas=True)
+    model = load_archive(archived_model_file).model
+    num_outputs = 0
+    num_sentences = 0
+    with open(output_file, "w") as outfile:
+        for line in open(input_file):
+            num_sentences += 1
+            input_data = json.loads(line)
+            sentence = input_data["sentence"]
+            structured_representations = input_data["worlds"]
+            labels = input_data["labels"]
+            instance = reader.text_to_instance(sentence, structured_representations)
+            # Tweaking the instance to add an extra field to coerce the model to generate a k-best
```

Comment:
Yeah, instead of tweaking the instance, just call `model._decoder_trainer.max_num_decoded_sequences = 100`, or something similar, after loading the model.  If you have this script reach down into the `_decoder_trainer`, the `Model` doesn't have to know about the parameter at all.  Given that this is a pretty rare use case, I don't think it's a big deal to just do `# pylint: ignore=protected-access` on that line where you set that parameter.